core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -223,18 +223,20 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging with RawlsInstrum
         )
       }
 
-    def getWorkspaceAndSubmitter(dataAccess: DataAccess): ReadWriteAction[(RawlsUserEmail, WorkspaceRecord)] =
+    def getWorkspace(dataAccess: DataAccess, workspaceId: UUID): ReadWriteAction[WorkspaceRecord] =
       for {
-        submissionRec <- dataAccess.submissionQuery.findById(submissionId).result.map(_.head)
-        workspaceRec <- dataAccess.workspaceQuery.findByIdQuery(submissionRec.workspaceId).result.map(_.head)
-      } yield (RawlsUserEmail(submissionRec.submitterEmail), workspaceRec)
+        workspaceRec <- dataAccess.workspaceQuery.findByIdQuery(workspaceId).result.map(_.head)
+      } yield workspaceRec
 
-    def abortActiveWorkflows(submissionId: UUID): Future[Seq[(Option[String], Try[ExecutionServiceStatus])]] =
+    def abortActiveWorkflows(
+      submissionRec: SubmissionRecord
+    ): Future[Seq[(Option[String], Try[ExecutionServiceStatus])]] =
       datasource.inTransaction { dataAccess =>
         for {
           // look up abortable WorkflowRecs for this submission
           wfRecs <- dataAccess.workflowQuery.findWorkflowsForAbort(submissionId).result
```

Comment:
Just checking if I'm reading this right - looks like `submissionId` used to be passed in to the method, but now we have an entire record passed in, so wouldn't this need to be updated to `submissionRec.Id`?

Reply:
Good question - but I don't think this needs to change. `SubmissionMonitorActor` accepts a `submissionId` in its constructor, so this line of code changed from using the `submissionId` argument to using the `submissionId` class-level variable. Previously, the class-level variable IS what was passed as the argument.

Reply:
heh yes, the `submissionId` argument was unnecessary before. Turning `submissionRec` into an instance variable is less ideal … since this actor operates repeatedly over time for the same submission, the submission _id_ is constant, but other values in `submissionRec` can and do change.